### PR TITLE
Hive: Fix some message typos in HiveCatalog: Matastore => Metastore

### PR DESCRIPTION
**Pull request by szehon-ho**
Noticed a lot of these errors while trying to debug timeouted Hive tests

### CODE REVIEW

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java [hunk 1]
```diff
@@ -251,12 +251,12 @@ public void createNamespace(Namespace namespace, Map<String, String> meta) {
             namespace);
 
     } catch (TException e) {
-      throw new RuntimeException("Failed to create namespace " + namespace + " in Hive MataStore", e);
+      throw new RuntimeException("Failed to create namespace " + namespace + " in Hive MetaStore", e);
```
Comment (aokolnychyi) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:254]: Out of curiosity, how do Hive folks prefer to write it? `MetaStore` or `Metastore`?

**Review by aokolnychyi — APPROVED** [review]

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java (hunk 1 above)
Comment (szehon-ho) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:254]: I honestly always used Metastore and it's like that in most of the Hive docs, but I've seen the camel case a lot too.

+ @cwsteinbach  @pvary  ?

**Review by rdblue — APPROVED** [review]

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java (hunk 1 above)
Comment (kbendick) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:254]: I’ve also always used `Metastore`.

Looking at this Hive design doc, it’s also `Metastore` with no capital S. I think this makes the most sense as it’s not really two words imo (and several Hive docs confirm it as being one word): https://cwiki.apache.org/confluence/plugins/servlet/mobile?contentId=27362072#Design-Metastore

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java [hunk 2]
```diff
@@ -251,12 +251,12 @@ public void createNamespace(Namespace namespace, Map<String, String> meta) {
...
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException(
-          "Interrupted in call to createDatabase(name) " + namespace + " in Hive MataStore", e);
+          "Interrupted in call to createDatabase(name) " + namespace + " in Hive MetaStore", e);
```
Comment (kbendick) [inline hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java:259]: If we go with `Metastore` and not `MetaStore` (which the former is correct according to a number of Hive and Cloudera docs), this also needs to be updated - plus several other locations :)

**Review by kbendick — COMMENTED** [review]
Thanks for doing this. I’m +0, only because I really feel it’s `Metastore` as one word and not `MetaStore` as two words.

I’ll let others chime in on that, but a basic search of docs etc seem to bring up Metastore as one word much more often than I’ve seen “official” docs using the phrase as two words.

File: hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java (hunk 1 above)
Comment (rdblue) [reply hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java]: I'd say Metastore. Capitalizing the S was to match the class name.

**Review by rdblue — APPROVED** [review]

**Comment by szehon-ho** [comment]
Rebase to re-trigger test

**Comment by szehon-ho** [comment]
Test failure seemed flaky, try to fix it in : https://github.com/apache/iceberg/pull/2989 , rebase to trigger again

**Review by pvary — APPROVED** [review]

**Pull request merged**